matrix-synapse-riot/terraform/matrix-server/dehydrated-hooks

```diff
@@ -0,0 +1,214 @@
+    #   TXT record. For HTTP validation it is the value that is expected
+    #   be found in the $TOKEN_FILENAME file.
+
+    # Simple example: Use nsupdate with local named
+    # printf 'server 127.0.0.1\nupdate add _acme-challenge.%s 300 IN TXT "%s"\nsend\n' "${DOMAIN}" "${TOKEN_VALUE}" | nsupdate -k /var/run/named/session.key
+    DO_HOSTNAME=$(echo $DOMAIN | sed -e "s/$DO_DOMAIN//" | sed 's/.$//')
+    echo "Issueing channenge for $DO_HOSTNAME Lets Encrypt domain is $DOMAIN Challenge is $TOKEN_VALUE"
+    DNS_ID=$(curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer $DO_TOKEN" -d '{"type":"TXT","name":"_acme-challenge.'$DO_HOSTNAME'","data":"'$TOKEN_VALUE'","priority":null,"port":null,"ttl":60,"weight":null,"flags":null,"tag":null}' "https://api.digitalocean.com/v2/domains/$DO_DOMAIN/records" | jq -r '.domain_record.id')
+    echo "$DOMAIN $DNS_ID">> domainstate.txt
+    echo "cat domainstate.txt"
+    cat domainstate.txt
+}
+
+clean_challenge() {
+    local DOMAIN="${1}" TOKEN_FILENAME="${2}" TOKEN_VALUE="${3}"
+
+    # This hook is called after attempting to validate each domain,
+    # whether or not validation was successful. Here you can delete
+    # files or DNS records that are no longer needed.
+    #
+    # The parameters are the same as for deploy_challenge.
+
+    # Simple example: Use nsupdate with local named
+    # printf 'server 127.0.0.1\nupdate delete _acme-challenge.%s TXT "%s"\nsend\n' "${DOMAIN}" "${TOKEN_VALUE}" | nsupdate -k /var/run/named/session.key
+    DNS_ID=$(grep -E '(^| )'$DOMAIN'( |$)' domainstate.txt | awk -F " " '{ print $2 }')
+    echo "Deleteing $DOMAIN ID is $DNS_ID"
+    curl -X DELETE -H "Content-Type: application/json" -H "Authorization: Bearer $DO_TOKEN" "https://api.digitalocean.com/v2/domains/$DO_DOMAIN/records/$DNS_ID"
+}
+
+deploy_cert() {
+    local DOMAIN="${1}" KEYFILE="${2}" CERTFILE="${3}" FULLCHAINFILE="${4}" CHAINFILE="${5}" TIMESTAMP="${6}"
+
+    # This hook is called once for each certificate that has been
+    # produced. Here you might, for instance, copy your new certificates
+    # to service-specific locations and reload the service.
+    #
+    # Parameters:
+    # - DOMAIN
+    #   The primary domain name, i.e. the certificate common
+    #   name (CN).
+    # - KEYFILE
+    #   The path of the file containing the private key.
+    # - CERTFILE
+    #   The path of the file containing the signed certificate.
+    # - FULLCHAINFILE
+    #   The path of the file containing the full certificate chain.
+    # - CHAINFILE
+    #   The path of the file containing the intermediate certificate(s).
+    # - TIMESTAMP
+    #   Timestamp when the specified certificate was created.
+
+    # Simple example: Copy file to nginx config
+    if [ ! -d /etc/nginx/ssl/$DOMAIN/ ]
+    then
+    	mkdir -m  700 /etc/nginx/ssl/$DOMAIN/
+    fi
+    cp "${KEYFILE}" "${FULLCHAINFILE}" /etc/nginx/ssl/$DOMAIN/ ; chown -R root:root /etc/nginx/ssl/
+    # systemctl reload nginx
```

Comment:
I think it is better to keep their example unmodified and all together, as in other parts:

```
    # Simple example: Copy file to nginx config
    # cp "${OCSPFILE}" /etc/nginx/ssl/; chown -R nginx: /etc/nginx/ssl
    # systemctl reload nginx
```

Then put our custom stuff. Looking at this now I am quite confused.